### ISSUE TITLE
${callsite} value is <no type>.lambda_method

### ISSUE THREAD

**Issue by alataffective** [NLog/NLog#2087]
I am migrating an ASP.NET Web Forms project to ASP.NET Core (.NET Framework). I am logging to a database. In the previous project, using ${callsite:skipFrames=1} produced the method name. However in ASP.NET Core I get a value of <no type>.lambda_method,

**Type**: Question, possibly bug.
**Version**:
  NLog.Web.AspNetCore: 4.3.1
  NLog.Extensions.Logging: 1.0.0-rtm-beta4
**Platform**: .NET Framework 4.6.2

**Current NLog config**

```xml
<?xml version="1.0" encoding="utf-8" ?>
<nlog 
  xmlns="http://www.nlog-project.org/schemas/NLog.xsd"
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://www.nlog-project.org/schemas/NLog.xsd NLog.xsd"
  autoReload="true"
  throwExceptions="false"
  internalLogLevel="Off" 
  internalLogFile="c:\temp\nlog-internal.log">

  <targets>
      <target
        name="database"
        xsi:type="Database"
        commandType="StoredProcedure"
        commandText="dbo.WriteLog" > <!-- #security Limit permissions -->
        <parameter name="@MachineName"    layout="${machinename}" />
        <parameter name="@SiteName"       layout="${iis-site-name}" />
        <parameter name="@DateTime"       layout="${date}" />
        <parameter name="@Level"          layout="${level}" />
        <parameter name="@Username"       layout="${aspnet-user-identity}" />
        <parameter name="@Message"        layout="${message}" />
        <parameter name="@Logger"         layout="${logger}" />
        <parameter name="@Properties"     layout="${all-event-properties:separator=|}" />
        <parameter name="@ServerName"     layout="${aspnet-request:serverVariable=SERVER_NAME}" />
        <parameter name="@Port"           layout="${aspnet-request:serverVariable=SERVER_PORT}" />
        <parameter name="@Url"            layout="${aspnet-request:serverVariable=HTTP_URL}" />
        <parameter name="@Https"          layout="${when:inner=1:when='${aspnet-request:serverVariable=HTTPS}' == 'on'}${when:inner=0:when='${aspnet-request:serverVariable=HTTPS}' != 'on'}" />
        <parameter name="@ServerAddress"  layout="${aspnet-request:serverVariable=LOCAL_ADDR}" />
        <parameter name="@RemoteAddress"  layout="${aspnet-request:serverVariable=REMOTE_ADDR}:${aspnet-request:serverVariable=REMOTE_PORT}" />
        <parameter name="@CallSite"       layout="${callsite:skipFrames=1}" />
        <parameter name="@Exception"      layout="${exception:tostring}" />
      </target>
  </targets>

  <rules>
    <logger name="*" minlevel="Info" writeTo="database" />
  </rules>
</nlog>

```

**Current result**
`<no type>`.lambda_method

**Expected result**
(Method name)

**Workarounds**
None found.

**Known working version**
NLog 4

I have tried different values for skipFrames (including complete removal) but without success.

**Comment by 304NotModified** [comment]
could you output the info when using 

   ${stacktrace:format=raw:topFrames=10}

Is your method in this output?

**Comment by alataffective** [comment]
Yes, it is. The calling method is AccountController.Register (see below).

InvokeNextActionFilterAsync at offset 106 in file:line:column <filename unknown>:0:0
Start<TStateMachine> at offset 67 in file:line:column <filename unknown>:0:0
MoveNext at offset 119 in file:line:column <filename unknown>:0:0
Next at offset 3353 in file:line:column <filename unknown>:0:0
InvokeActionMethodAsync at offset 106 in file:line:column <filename unknown>:0:0
Start<TStateMachine> at offset 67 in file:line:column <filename unknown>:0:0
MoveNext at offset 1745 in file:line:column <filename unknown>:0:0
Execute at offset 15 in file:line:column <filename unknown>:0:0
lambda_method at offset 104 in file:line:column <filename unknown>:0:0
**Register** at offset 102 in file:line:column <filename unknown>:0:0

**Comment by 304NotModified** [comment]
Could you please post your NLog version? (Screenshot of NLog.DLL properties or assembly version) Thanks!

**Comment by alataffective** [comment]
![nlog](https://cloud.githubusercontent.com/assets/18467932/25793096/6605f2ae-33c2-11e7-8d6a-6a42ce8519ea.png)

**Comment by 304NotModified** [comment]
OK thanks, this is indeed a bug then. We have to dig into the code. 

We skip "movenext" here: https://github.com/NLog/NLog/blob/coreCLR/src/NLog/LoggerImpl.cs#L163

Probably the type is wrong: https://github.com/NLog/NLog/blob/coreCLR/src/NLog/LoggerImpl.cs#L170

Is think something you could debug for us?

**Comment by 304NotModified** [comment]
also could you give us the calling side? (including the lambda and async etc)? We should add this to the unit test cases.

**Comment by alataffective** [comment]
Sorry, the error is mine. I've just looked at this again and if I remove skipFrames=1 so that the layout value is just ${callsite}, all is well. The skipFrames modifier was brought in from the older code where I had a wrapper around the logger. I tried this before and concluded that it did not fix it, but it's possible I was tripped up by an AsyncWrapper around the writes to the database and was perhaps looking at the wrong log entry? Not sure. Anyway, thanks for your time.
